src/ImageStorage/Image/Transformation/WatermarkRand.php

```diff
@@ -0,0 +1,168 @@
+<?php
+namespace ImageStorage\Image\Transformation;
+
+class WatermarkRand implements \ImageStorage\Image\Transformation
+{
+	/**
+	 * @var \ImageStorage\Image\Structure\Image
+	 */
+	private $_imageStruct;
+
+	/**
+	 * @var \ImageStorage\Image\Structure\WatermarkRand|null
+	 */
+	private $_watermarkRand = null;
+
+	/**
+	 * @param \ImageStorage\Image\Structure\WatermarkRand $watermark
+	 *
+	 * @throws \Exception
+	 */
+	public function __construct(\ImageStorage\Image\Structure\WatermarkRand $watermarkRand = null)
+	{
+		if (!($watermarkRand instanceof \ImageStorage\Image\Structure\WatermarkRand))
```

Comment:
What’s the point of this exception? Couldn’t you do the same using just the method signature?